src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -571,7 +571,12 @@ export class SearchSource {
           }
         });
       }),
-      map((response) => onResponse(searchRequest, response, options))
+      map((response) => {
+        if (response.isPartial) {
```

Comment:
Let's reuse these helpers:
https://github.com/elastic/kibana/blob/77aecd598421b52a485cbc2e790f767435a34139/src/plugins/data/common/search/utils.ts#L13-L32

also, see how we should check `isRunning` also